food-truck-frontend/src/pages/truck/edit/[truckId].tsx

```diff
@@ -0,0 +1,114 @@
+import React, { Component } from 'react'
+import {Button, TextField, Typography} from "@material-ui/core";
+import api from "../../../util/api";
+import {AxiosResponse} from "axios";
+import Router, {useRouter} from "next/router";
+import Form from "../../../components/Form";
+import NotFound from "../../../components/NotFound";
+
+interface EditTruckState {
```

Comment:
Is this a carbon copy of the state from Truck view? If they're guaranteed to need the same information to create and edit trucks, you could refactor it into its own type (`TruckManagementState`) and then redefine these states as an intersection of types (`EditTruckState = TruckMangementState | { /* custom stuff */ };`